crates/core/plugin_sm/src/operation_logs.rs

```diff
@@ -75,8 +91,10 @@ impl OperationLogs {
     }
 
     pub fn remove_outdated_logs(&self) -> Result<(), OperationLogsError> {
+        // TODO:
```

Comment:
not yet done because it needs a refactor. I am currently working on it. It has been described in the body of the PR in the "what is yet missing" section, but I could have made it more clear